src/frontend/game/components/ChangeName.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'preact/hooks';
+import { type FunctionalComponent } from 'preact';
+import { LineInput } from 'src/frontend/components/LineInput.tsx';
+import { useClientContext } from 'src/frontend/game/ClientContextProvider.tsx';
+import { validName } from 'src/agnostic/validation.ts';
+
+export const ChangeName: FunctionalComponent = () => {
+	let [clientContext, dispatchGameEvent] = useClientContext();
+	const myId = clientContext.clientState.player.id;
+	const me = clientContext.gameState.players[myId];
+	let [inputValue, setInputValue] = useState(me.name);
+
+	// can only change name if new name valid and
+	// different from current name
+	const canChangeName = validName(inputValue) && me.name !== inputValue;
+
+	const onClick = () => {
+		dispatchGameEvent({
+			type: 'change-player-name',
+			data: {
+				id: myId,
+				name: inputValue,
+			},
+		});
```

Comment:
we create an instance of the event we defined earlier, and dispatch it to the server, which will echo it back down to this player if it's valid, and then the clientContext will update and the app will re-render